contrib/cctz-cmake/CMakeLists.txt

```diff
@@ -60,574 +60,17 @@ if (NOT EXTERNAL_CCTZ_LIBRARY_FOUND OR NOT EXTERNAL_CCTZ_LIBRARY_WORKS)
-            Indian/Cocos
-            Indian/Comoro
-            Indian/Kerguelen
-            Indian/Mahe
-            Indian/Maldives
-            Indian/Mauritius
-            Indian/Mayotte
-            Indian/Reunion
-            Iran
-            Israel
-            Jamaica
-            Japan
-            Kwajalein
-            Libya
-            MET
-            Mexico/BajaNorte
-            Mexico/BajaSur
-            Mexico/General
-            MST
-            MST7MDT
-            Navajo
-            NZ
-            NZ-CHAT
-            Pacific/Apia
-            Pacific/Auckland
-            Pacific/Bougainville
-            Pacific/Chatham
-            Pacific/Chuuk
-            Pacific/Easter
-            Pacific/Efate
-            Pacific/Enderbury
-            Pacific/Fakaofo
-            Pacific/Fiji
-            Pacific/Funafuti
-            Pacific/Galapagos
-            Pacific/Gambier
-            Pacific/Guadalcanal
-            Pacific/Guam
-            Pacific/Honolulu
-            Pacific/Johnston
-            Pacific/Kiritimati
-            Pacific/Kosrae
-            Pacific/Kwajalein
-            Pacific/Majuro
-            Pacific/Marquesas
-            Pacific/Midway
-            Pacific/Nauru
-            Pacific/Niue
-            Pacific/Norfolk
-            Pacific/Noumea
-            Pacific/Pago_Pago
-            Pacific/Palau
-            Pacific/Pitcairn
-            Pacific/Pohnpei
-            Pacific/Ponape
-            Pacific/Port_Moresby
-            Pacific/Rarotonga
-            Pacific/Saipan
-            Pacific/Samoa
-            Pacific/Tahiti
-            Pacific/Tarawa
-            Pacific/Tongatapu
-            Pacific/Truk
-            Pacific/Wake
-            Pacific/Wallis
-            Pacific/Yap
-            Poland
-            Portugal
-            PRC
-            PST8PDT
-            ROC
-            ROK
-            Singapore
-            Turkey
-            UCT
-            Universal
-            US/Alaska
-            US/Aleutian
-            US/Arizona
-            US/Central
-            US/Eastern
-            US/East-Indiana
-            US/Hawaii
-            US/Indiana-Starke
-            US/Michigan
-            US/Mountain
-            US/Pacific
-            US/Samoa
-            UTC
-            WET
-            W-SU
-            Zulu)
-
+        # get the list of timezones from tzdata shipped with cctz
         set(TZDIR ${LIBRARY_DIR}/testdata/zoneinfo)
+        file(STRINGS ${LIBRARY_DIR}/testdata/version TZDATA_VERSION)
+        set_property(GLOBAL PROPERTY TZDATA_VERSION_PROP "${TZDATA_VERSION}")
+        message(STATUS "Packaging with tzdata version: ${TZDATA_VERSION}")
+
         set(TZ_OBJS)
 
+        # each file/symlink in that dir (except of tab and localtime) store the info about timezone
+        execute_process(COMMAND bash -c "cd ${TZDIR} && find * -type f,l -and ! -name '*.tab' -and ! -name 'localtime' | sort | paste -sd ';'" OUTPUT_VARIABLE TIMEZONES )
```

Comment:
Not sure if it's good: from one side it's better to make it explicit, but from the other side it's too easy to forget update the list, and some timezones supported by tzdb wll not work. 

Right now we had those missing:
```
America/Nuuk
Etc/GMT+0
Etc/GMT-0
Etc/GMT0
Etc/GMT+1
Etc/GMT-1
Etc/GMT+10
Etc/GMT-10
Etc/GMT+11
Etc/GMT-11
Etc/GMT+12
Etc/GMT-12
Etc/GMT-13
Etc/GMT-14
Etc/GMT+2
Etc/GMT-2
Etc/GMT+3
Etc/GMT-3
Etc/GMT+4
Etc/GMT-4
Etc/GMT+5
Etc/GMT-5
Etc/GMT+6
Etc/GMT-6
Etc/GMT+7
Etc/GMT-7
Etc/GMT+8
Etc/GMT-8
Etc/GMT+9
Etc/GMT-9
GMT+0
GMT-0
```

Reply:
Also - not sure if TZ with plus / minus char in name will work properly from binary

Reply:
See also 
https://github.com/ClickHouse/ClickHouse/issues/4443
https://github.com/ClickHouse/ClickHouse/issues/4421

Reply:
Autogeneration is good, I wanted to do it initially.

About GMT+- - I expect they won't work - then we should remove them from the list to avoid keeping them as trash.

Reply:
It's better to get the list of timezones from CCTZ, otherwise it's too easy to build ClickHouse with outdated tzdata from system. People tend to not update the OS packages when not necessarily.

Or: you can somehow decide what is newer.

Reply:
Yes, you have to remove some
```
2020-08-12 15:52:52 ld.lld-10: error: duplicate symbol: _binary_Etc_GMT_0_start
2020-08-12 15:52:52 >>> defined at Etc_GMT+0.o:(.rodata+0x0) in archive /ClickHouse/build/contrib/cctz-cmake/libtzdata.a
2020-08-12 15:52:52 >>> defined at Etc_GMT-0.o:(.rodata+0x0) in archive /ClickHouse/build/contrib/cctz-cmake/libtzdata.a
```

Reply:
> About GMT+- - I expect they won't work -

They work if tzdata installed in system:
https://github.com/ClickHouse/ClickHouse/blob/0f79eb3cc518977a46b8020c6e317cae15ff27b5/base/common/DateLUTImpl.cpp#L218-L219

 WIll try to fix them (want to have them working properly from binary).

> It's better to get the list of timezones from CCTZ

Agree and it's already done this way (so updating tzdata package is not needed and i removed that).

So we only need to remember about updating cctz itself (i've added a simple bash script to check if update is needed - you can add it in CI).